File: random6.py
import random

random_number = random.randint(1, 100)
counter = 6

user_number = int(input("Enter a whole number between 1 and 100: "))

for i in range(1, 6):
  if user_number == random_number:
    print("You guessed it! Congratulations!")
    print(r"""
   _____                            _       _ 
  / ____|                          | |     | |
 | |     ___  _ __   __ _ _ __ __ _| |_ ___| |
 | |    / _ \| '_ \ / _` | '__/ _` | __/ __| |
 | |___| (_) | | | | (_| | | | (_| | |_\__ \_|
  \_____\___/|_| |_|\__, |_|  \__,_|\__|___(_)
                     __/ |                    
                    |___/                     
    """)
    break
  elif user_number > random_number:
    print("Too big! Try again.")
    counter -= 1
    user_number = int(float(input("Remaining attempts: " + str(counter) + ".\nEnter a whole number between 1 and 100: ")))
  elif user_number < random_number:
    print("Too small! Try again.")
    counter -= 1
    user_number = int(float(input("Remaining attempts: " + str(counter) + ".\nEnter a whole number between 1 and 100: ")))

if user_number != random_number:
  print("The random number was " + str(random_number) + ". Better luck next time!")

File: random6.js
let randomNumber = Math.round(Math.random() * 100);
let counter = 6;

function userInput(prompt) {
  let readlineSync = require('readline-sync');
  let input = readlineSync.question(prompt);
  return input;
}

let userNumber = Math.round(userInput("Enter a whole number between 1 and 100: "));

for (let i = 0; i < 5; i++) {
  if (parseInt(userNumber) > randomNumber) {
    console.log("Too big! Try again.");
    counter -= 1;
    userNumber = Math.round(userInput(`Remaining attempts: ${counter}.\nEnter a whole number between 1 and 100: `));
  } else if (parseInt(userNumber) < randomNumber) {
    console.log("Too small! Try again.");
    counter -= 1;
    userNumber = Math.round(userInput(`Remaining attempts: ${counter}.\nEnter a whole number between 1 and 100: `));
  } else if (parseInt(userNumber) === randomNumber) {
    console.log("You guessed it! Congratulations!");
    return randomNumber;
  } else {
    console.log("Please enter a whole number between 1 and 100.");
    userNumber = Math.round(userInput(`Remaining attempts: ${counter}.\nEnter a whole number between 1 and 100: `));
  }
}

if (userNumber != randomNumber) {
  console.log(`The random number was ${randomNumber}. Better luck next time!`);
}
